src/main/java/emu/grasscutter/server/game/GameSession.java

```diff
@@ -283,7 +289,36 @@ public void onMessage(ChannelHandlerContext ctx, ByteBuf data) {
 			packet.release();
 		}
 	}
-	
+
+	private void updateLag() {
+		final UkcpServerChildChannel channel = (UkcpServerChildChannel) super.getChannel();
+		final Set<Field> fields = ReflectionUtils.getFields(UkcpServerChildChannel.class, v -> "ukcp".equals(v.getName()));
+		if (!fields.isEmpty()) {
+			try {
+				Ukcp ukcp = (Ukcp) getFieldValue(channel, fields);
+				final Set<Field> fieldsOfUkcp = ReflectionUtils.getFields(Ukcp.class, v -> "kcp".equals(v.getName()));
+				if (!fieldsOfUkcp.isEmpty()) {
+					final Kcp kcp = (Kcp) getFieldValue(ukcp, fieldsOfUkcp);
+					final KcpMetric metric = kcp.getMetric();
+					final int srtt = metric.srtt();
+					this.lag = srtt / 2;
+				}
+			} catch (Exception e) {
+
+			}
+		}
+	}
+
+	private Object getFieldValue(Object ukcp, Set<Field> fieldsOfUkcp) throws IllegalAccessException {
+		final Field kcpField = fieldsOfUkcp.iterator().next();
+		kcpField.setAccessible(true);
+		return kcpField.get(ukcp);
+	}
+
+	public int getLag() {
```

Comment:
Getting lag this way is unnessecary, you can just get it from the player ping if you really need to.

Reply:
I found the PingReq was ticking every five seconds, so it may not represent in-time lag. I notice that one contributor is replacing the kcp lib. Maybe I can get the lag more gracefully or won't count the lag at all.